src/main/java/uk/gov/digital/ho/hocs/domain/service/ProcessingService.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.digital.ho.hocs.domain.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.application.LogEvent;
+import uk.gov.digital.ho.hocs.domain.exceptions.ApplicationExceptions;
+import uk.gov.digital.ho.hocs.domain.queue.common.MessageHandler;
+
+import javax.validation.constraints.NotNull;
+import java.time.LocalDateTime;
+
+@Service
+@Slf4j
+public class ProcessingService {
+
+    private final MessageLogService messageLogService;
+
+    private final MessageHandler messageHandler;
+
+    public ProcessingService(MessageLogService messageLogService, MessageHandler messageHandler) {
+        this.messageLogService = messageLogService;
+        this.messageHandler = messageHandler;
+    }
+
+    public void processMessages(int maxMessages, LocalDateTime from, @NotNull LocalDateTime to) {
+        var messageCount = messageLogService.getCountOfPendingMessagesBetweenDates(from, to);
+        if (messageCount > maxMessages) {
+            throw new ApplicationExceptions.TooManyMessagesException(
```

Comment:
I feel like we should have some form of circuit breaker here, maybe even a database table with the configuration (although seems overkill).